src/runtime/metal.cpp

```diff
@@ -591,22 +601,27 @@ WEAK int halide_metal_copy_to_host(void *user_context, buffer_t* buffer) {
     device_copy c = make_device_to_host_copy(buffer);
     uint8_t *device_ptr = (uint8_t *)buffer_contents((mtl_buffer *)c.src);
 
-    // TODO: Is this 32-bit or 64-bit? Leaving signed for now
-    // in case negative strides.
-    for (int w = 0; w < (int)c.extent[3]; w++) {
-        for (int z = 0; z < (int)c.extent[2]; z++) {
-            for (int y = 0; y < (int)c.extent[1]; y++) {
-                for (int x = 0; x < (int)c.extent[0]; x++) {
-                    uint64_t off = (x * c.stride_bytes[0] +
-                                    y * c.stride_bytes[1] +
-                                    z * c.stride_bytes[2] +
-                                    w * c.stride_bytes[3]);
-                    void *src = device_ptr + off;
-                    void *dst = (void *)(c.dst + off);
-                    memcpy(dst, src, c.chunk_size);
-                }
-            }
-        }
+    // If this is a zero copy buffer, these pointers will be the same.
+    if ((uint8_t *)c.dst != device_ptr) {
+	// TODO: Is this 32-bit or 64-bit? Leaving signed for now
+	// in case negative strides.
+	for (int w = 0; w < (int)c.extent[3]; w++) {
```

Comment:
seems like this should be a good candidate for pulling into a helper function somwhere; near identical code is just a bit earlier.